tests/test_regularization.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python3
+
+import random
+
+import numpy as np
+import pandas as pd
+import pytest
+import torch
+from utils.dataset_generators import generate_event_dataset, generate_holiday_dataset
+
+from neuralprophet import NeuralProphet, df_utils
+from neuralprophet.utils import reg_func_abs
+
+# Fix random seeds
+torch.manual_seed(0)
+random.seed(0)
+np.random.seed(0)
+
+# Variables
+EPOCHS = 10
+BATCH_SIZE_HOLIDAYS = 32
+BATCH_SIZE_EVENTS = 1
+LEARNING_RATE = 0.1
+REGULARIZATION = 10
+# Map holiday name to a y value for dataset generation
+Y_HOLIDAYS_OVERRIDE = {
+    "Washington's Birthday": 1,
+    "Labor Day": 5,
+    "Christmas Day": 1,
+}
+Y_EVENTS_OVERRIDE = {
+    "2022-01-13": 1,
+}
+
+
+def test_reg_func_abs():
+    assert pytest.approx(1) == reg_func_abs(torch.Tensor([1]))
+    assert pytest.approx(0) == reg_func_abs(torch.Tensor([0]))
+    assert pytest.approx(1) == reg_func_abs(torch.Tensor([-1]))
+
+    assert pytest.approx(1) == reg_func_abs(torch.Tensor([1, 1, 1]))
+    assert pytest.approx(0) == reg_func_abs(torch.Tensor([0, 0, 0]))
+    assert pytest.approx(1) == reg_func_abs(torch.Tensor([-1, -1, -1]))
+
+    assert pytest.approx(0.6666666) == reg_func_abs(torch.Tensor([-1, 0, 1]))
+    assert pytest.approx(20) == reg_func_abs(torch.Tensor([-12, 4, 0, -1, 1, 102]))
+
+
+def test_regularization_holidays():
+    df = generate_holiday_dataset(y_holidays_override=Y_HOLIDAYS_OVERRIDE)
+    df = df_utils.check_dataframe(df, check_y=False)
+
+    m = NeuralProphet(
+        yearly_seasonality=False,
+        weekly_seasonality=False,
+        daily_seasonality=False,
+        growth="off",
+        epochs=EPOCHS,
+        batch_size=BATCH_SIZE_HOLIDAYS,
+        learning_rate=LEARNING_RATE,
+    )
+    m = m.add_country_holidays("US", regularization=REGULARIZATION)
+    m.fit(df, freq="D")
+
+    for country_holiday in m.country_holidays_config.holiday_names:
+        event_params = m.model.get_event_weights(country_holiday)
+        weight_list = [param.detach().numpy() for _, param in event_params.items()]
+        if country_holiday in Y_HOLIDAYS_OVERRIDE.keys():
+            assert weight_list[0] <= 0.1
+        else:
+            assert weight_list[0] <= 0.5
```

Comment:
When regularizing, we still want to have the meaningful weights to not be reduced too much.
I think that this ought to be: `assert weight_list[0] > 0.5`?

Reply:
The original issue reported (#133) was that weights are not reduced to zero with regularization enabled. Happy to adjust the test case, but then we cannot be sure that regularization does actually reduce the weights in a meaningful way anymore. That's also why I added two test cases for each component (events and holidays), to make sure regularization has a meaningful impact on weights. Let me know your thoughts.

To ensure we still have meaningful weights in contrast to the meaningless weights we could add a check for the meaningful ones to have `weight_list[0] > 0.1`?